sdk/tables/azure-data-tables/tests/test_table_client.py

```diff
@@ -223,30 +224,262 @@ def test_client_with_sas_token(self, tables_storage_account_name, tables_primary
             expiry=datetime.utcnow() + timedelta(hours=1),
         )
 
-        with TableServiceClient(base_url, credential=AzureSasCredential(sas_token)) as client:
+        with TableClient.from_table_url(
+            f"{base_url}/{table_name}", credential=tables_primary_storage_account_key
+        ) as client:
+            table = client.create_table()
+            assert table.name == table_name
+
+        conn_str = f"AccountName={tables_storage_account_name};AccountKey={tables_primary_storage_account_key.named_key.key};EndpointSuffix=core.windows.net"
+        with TableClient.from_connection_string(conn_str, table_name) as client:
+            entities = client.query_entities(
+                query_filter="PartitionKey eq @pk",
+                parameters={"pk": "dummy-pk"},
+            )
+            for e in entities:
+                pass
+
+        with TableClient(
+            f"{base_url}/?{sas_token}", table_name, credential=tables_primary_storage_account_key
```

Comment:
Can we add some kind of request validation to these to ascertain which auth is being used? Something like this should work:
```python
def check_request_auth(pipeline_request):
    assert pipeline_request.http_request.url == f"{base_url}/?{sas_token}"
    assert pipeline_request.http_request.headers.get('Authorization') is not None

entities = client.quert_entities(..., raw_request_hook=check_request_auth)
```